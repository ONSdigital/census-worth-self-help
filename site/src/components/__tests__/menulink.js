import React from "react"
import renderer from "react-test-renderer"
import MenuLink from "../menulink"
import { render, fireEvent } from "react-testing-library"

describe("MenuLink", () => {
  let hidden_nodes = [
		{title:'child1', link:'child1', children:[]}, {title:'child2', link:'child2',
			 children: [{title:'grandchild1', link:'grandchild1'}, {title:'grandchild2', link:'grandchild2'}]
			}
  ]

  it("renders correctly without hidden nodes", () => {
    const tree = renderer
      .create(<MenuLink title="basic" link="basic" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly with hidden nodes", () => {
    const tree = renderer
      .create(<MenuLink title="basic" link="basic" hidden_nodes={hidden_nodes}/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("toggling hidden nodes", () => {
  	  // Click the toggle button, child container should appear and dissappear
  	const { getByTestId, getAllByTestId, queryByTestId } = render(<MenuLink title="basic" link="basic" hidden_nodes={hidden_nodes}/>);
  	
  	expect(queryByTestId('child-container')).toBeNull()

  	fireEvent.click(getByTestId('toggle-button'));

  	expect(getByTestId('child-container')).toBeDefined()

  	fireEvent.click(getAllByTestId('toggle-button')[0]);

  	expect(queryByTestId('child-container')).toBeNull()
  });
})