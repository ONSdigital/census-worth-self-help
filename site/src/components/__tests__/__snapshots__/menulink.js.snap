// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuLink renders correctly with hidden nodes 1`] = `
<div>
  <div
    css={
      Object {
        "map": undefined,
        "name": "1329ezx",
        "next": undefined,
        "styles": "
            
    margin: 15px 0px;
  ;
            display: flex;
          ",
      }
    }
  >
    <a
      css={
        Object {
          "map": undefined,
          "name": "dcy533",
          "next": undefined,
          "styles": "
              
    text-decoration: none;
    font-size: large;
    color: white;
  ;
              flex-grow: 1;
            ",
        }
      }
      href="basic"
    >
      basic
    </a>
    <div
      css={
        Object {
          "map": undefined,
          "name": "9d6hc1",
          "next": undefined,
          "styles": "
                padding: 0px 30px;
                
    text-decoration: none;
    color: inherit;
    font-size: larger;
    color: white;
  ;
              ",
        }
      }
      data-testid="toggle-button"
      onClick={[Function]}
    >
      <span>
        +
      </span>
    </div>
  </div>
</div>
`;

exports[`MenuLink renders correctly without hidden nodes 1`] = `
<div>
  <div
    css={
      Object {
        "map": undefined,
        "name": "1329ezx",
        "next": undefined,
        "styles": "
            
    margin: 15px 0px;
  ;
            display: flex;
          ",
      }
    }
  >
    <a
      css={
        Object {
          "map": undefined,
          "name": "dcy533",
          "next": undefined,
          "styles": "
              
    text-decoration: none;
    font-size: large;
    color: white;
  ;
              flex-grow: 1;
            ",
        }
      }
      href="basic"
    >
      basic
    </a>
  </div>
</div>
`;
